planning/behavior_path_planner/src/scene_module/start_planner/start_planner_module.cpp

```diff
@@ -833,27 +827,28 @@ std::vector<Pose> StartPlannerModule::searchPullOutStartPoses(
     }
 
     if (utils::checkCollisionBetweenFootprintAndObjects(
-          local_vehicle_footprint, *backed_pose, pull_out_lane_stop_objects,
+          local_vehicle_footprint, *backed_pose, stop_objects_in_pull_out_lanes,
           parameters_->collision_check_margin)) {
       break;  // poses behind this has a collision, so break.
     }
 
-    pull_out_start_pose.push_back(*backed_pose);
+    pull_out_start_pose_candidates.push_back(*backed_pose);
   }
-  return pull_out_start_pose;
+  return pull_out_start_pose_candidates;
 }
 
-bool StartPlannerModule::isOverlappedWithLane(
-  const lanelet::ConstLanelet & candidate_lanelet,
-  const tier4_autoware_utils::LinearRing2d & vehicle_footprint)
+PredictedObjects StartPlannerModule::filterStopObjectsInPullOutLanes(
+  const lanelet::ConstLanelets & pull_out_lanes, const double velocity_threshold) const
 {
-  for (const auto & point : vehicle_footprint) {
-    if (boost::geometry::within(point, candidate_lanelet.polygon2d().basicPolygon())) {
-      return true;
-    }
-  }
+  // filter for objects located in pull_out_lanes and moving at a speed below the threshold
+  const auto [objects_in_pull_out_lanes, others] =
+    utils::path_safety_checker::separateObjectsByLanelets(
+      *planner_data_->dynamic_object, pull_out_lanes,
+      utils::path_safety_checker::isPolygonOverlapLanelet);
+  const auto stop_objects_in_pull_out_lanes = utils::path_safety_checker::filterObjectsByVelocity(
+    objects_in_pull_out_lanes, velocity_threshold);
```

Comment:
better to filter with velocity for computational cost

Reply:
agree!

Reply:
changed in https://github.com/autowarefoundation/autoware.universe/pull/5529/commits/f7d177542fd6be4e3a34cc28e2324ceace72153a